mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearches.java

```diff
@@ -251,7 +252,7 @@ private boolean textContains(String value, MailboxMessage message) throws IOExce
     }
 
     private InputStream textHeaders(MailboxMessage message) throws MimeIOException, IOException {
-        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        ByteArrayOutputStream out = new ByteArrayOutputStream(Ints.checkedCast(message.getFullContentOctets()));
```

Comment:
No, re-allocation is your concern here or you don't need that tickets.
There's several solutions to that, to most straightforward is to derive ByteArrayOutputStream to gain access to the byte[] and check that the instance you create is the same than the one you get after going through the method call (see ExposedByteArrayOutputStream).
Please note that ByteArrayOutputStream actually copies the byte[] on toByteArray, probably something we could get rid of too.
Another solution is to come with another Stream implementation, javax.mail has one that allow sharing the underlying byte[], commons-io has one that chain byte[] and I'm sure there's much more in the wild.